hazelcast/include/hazelcast/client/internal/socket/BaseSocket.h

```diff
@@ -111,37 +118,51 @@ class BaseSocket : public hazelcast::client::socket
         do_read(std::move(connection));
     }
 
+    bool write(std::vector<byte> buffer) override
```

Comment:
```suggestion
    bool write(std::vector<byte> &&buffer) override
```

Reply:
As general, usage of pointer, reference or smart pointer mostly cause some obscure bugs. So unless there is a strong reason, I use value semantic. For this case, there is no difference. `buffer` will not be reused. So there is no copy. Btw, this buffer approximately `20bytes` so it can be ignored easily.